### ISSUE TITLE
Running Cortex analyzers using Python from software collections (SCL)

### ISSUE THREAD

**Issue by jezkerwin** [TheHive-Project/Cortex#121]
Running Cortex on Red Hat Enterprise Linux.

The Analyzers requirements were installed using 'python27' and 'rh-python36' from software collections (SCL) 

When running an Analyzers like VirusTotal I'm getting the following output

```
{
  "errorMessage": "Invalid output\nTraceback (most recent call last):\n  File \"VirusTotal/virustotal.py\", line 8, in <module>\n    from cortexutils.analyzer import Analyzer\nImportError: No module named cortexutils.analyzer\n",
  "input": null,
  "success": false
}
```

Now I think this is because Cortex isn't aware of the different python versions.

Can someone tell me where I could change a config file to refer to the python27 and rh-python36 versions?

Thanks.

**Comment by crackytsi** [comment]
no, you Need to install the cortexutils for your python Environment...
`pip install cortexutils`

**Comment by 3c7** [comment]
Ah, @crackytsi was faster. :)

**Comment by 3c7** [comment]
Addition: if python3 is invoked using the command `rh-python3` on your system, it could be useful to link it to /bin/python3, so rh-python36 is executed using `python3`. Another possibility is adding rh-python3 to the $PATH of the cortex user before any other python3 path, so it gets invoked instead.

**Comment by jezkerwin** [comment]
@crackytsi  I have installed cortexutils for my python environment. That's the problem. Cortex isn't using my python environment.

```
[root@cortex ~]# python
Python 2.7.5 (default, May 31 2018, 09:41:32)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import cortexutils
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named cortexutils
>>>
[root@cortex ~]# scl enable python27 bash
[root@cortex ~]# python
Python 2.7.13 (default, May 23 2018, 14:21:55)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import cortexutils
>>>
```

**Comment by 3c7** [comment]
Whats the output of `which python` after enabling python27 using scl? That would be the one that has to be set as default python interpreter for your cortex user. (Analogue: python3)

**Comment by jezkerwin** [comment]
The output is `/opt/rh/python27/root/usr/bin/python`

**Comment by 3c7** [comment]
So there are two possible solutions, I think:

1. run `scl enable python bash` automatically for the cortex user
2. prepend `/opt/rh/python27/root/usr/bin/python` to the cortex users $PATH
